pkg/operator/staticpod/controller/revision/deployment_controller.go

```diff
@@ -52,16 +54,18 @@ type RevisionController struct {
 func NewRevisionController(
 	targetNamespace string,
 	configMaps []string,
+	optionalConfigMaps []string,
```

Comment:
no.  Make the normal configmaps and secrets become a struct with an `optional` member.